### ISSUE TITLE
Variable.getUnits() returns string

### ISSUE THREAD

**Issue by MichaelClerx** [cellml/libcellml#202]
Hi all, the method `Variable.getUnits()` returns the name of the current units object (I see this makes sense from the current implementation, because only the name is stored in the variable).

What was the rationale for this? And since all other getX() methods return X, instead of X.getName(), will this not be a bit confusing?

**Comment by hsorby** [comment]
It would seem to me then that the method should be  `getUnitsName()`.

**Comment by nickerso** [comment]
This is because the units in this case is equivalent to the name - its the variable's units property, not the corresponding units object. The actual units object can only be retrieved once the variable is in a component which is in a model... The whole units/unit/units thing in CellML has always been confusing :)

I'm happy sticking with `getUnits()`.

**Comment by MichaelClerx** [comment]
Hmmm, so if I do this:

    m1 = Model()
    m1.defineTheUnitBiscuitsAsMetersPerSecond()
    m2 = Model()
    m2.defineTheUnitBiscuitsAsKilogramPerMol
    v = Variable()
    v.setUnits('biscuits')

then the units `v` is in are going to change dependent on the model I stick the variable in?

even worse, I could do

    v.setUnits(TheUnitBiscuitsAsAmperePerBecquerel)

and then, even though I was very exact in specifying v's unit, it would change without notice when added to a model

**Comment by MichaelClerx** [comment]
It makes sense to me that a unit name in a XML file derives its meaning from a parent. But in this object model we're allowing variables to exist without a defined parent...

**Comment by nickerso** [comment]
yep - the behaviour you describe is what I would currently expect. Actual units are a property of the model, references to units are properties of variables.

I'm not sure this is actually good :) but its how things are currently...I guess a question to ask is whether a user would ever want to do anything with a variable's units if the variable is not in a component which is in a model?

**Comment by jonc125** [comment]
As a user, I think I'd want to be able to define a variable with actual units (object) associated, and when adding the variable to a component the units would get added to the corresponding model if they weren't there already. Probably with an error if there was a name conflict, and an option to add them to the component instead.

**Comment by jonc125** [comment]
But it's OK if the low-level API doesn't support this - we can have a user-friendly layer building on the raw manipulation primitives.

**Comment by MichaelClerx** [comment]
Hmmm... In a meaningful use case where you give a variable a string unit, you must have some idea of what that unit string eventually is going to mean, right? So why can't you, instead of giving the string, query the model/component for the unit object and pass that in instead?

**Comment by nickerso** [comment]
Not always, you can easily (and we often do) make use of `flux_units` or `concentration_units` etc. where you don't necessarily care in your mathematics what the actual units are just that they are consistently used by some collection of variables. Its only when finally assembling those variables into components and making a model, which will likely import specific units from some units dictionary or similar. But the math should be valid across a range of potential flux or concentration units...

I can see value in @jonc125's suggestion (remembering that in CellML 2.0 there are no longer component-scope units), but agree that that is not something that needs to be in the currently low-level libCellML.
